File: frontend/src/app/pages/raporlar/rezervasyon-durum-dagilimi/rezervasyon-durum-dagilimi-rapor.dto.ts
export interface RezervasyonDurumDagilimiOzetDto {
    toplamRezervasyonSayisi: number;

    taslakSayisi: number;
    onayliSayisi: number;
    checkInTamamlandiSayisi: number;
    checkOutTamamlandiSayisi: number;
    iptalSayisi: number;

    gerceklesenRezervasyonSayisi: number;
    devamEdenRezervasyonSayisi: number;

    iptalOrani: number;
    gerceklesmeOrani: number;
    checkInOrani: number;
    checkOutOrani: number;

    toplamKisiSayisi: number;
    toplamGeceSayisi: number;
}

export interface RezervasyonDurumDagilimiDurumSatiriDto {
    durum: string;
    durumLabel: string;
    rezervasyonSayisi: number;
    kisiSayisi: number;
    geceSayisi: number;
    oran: number;
}

export interface RezervasyonDurumDagilimiOdaTipiSatiriDto {
    odaTipiId: number;
    odaTipiAdi: string;
    rezervasyonSayisi: number;
    iptalSayisi: number;
    gerceklesenSayisi: number;
    kisiSayisi: number;
    geceSayisi: number;
    iptalOrani: number;
    gerceklesmeOrani: number;
}

export interface RezervasyonDurumDagilimiRezervasyonDto {
    rezervasyonId: number;
    referansNo: string | null;
    misafirAdiSoyadi: string | null;

    girisTarihi: string;
    cikisTarihi: string;

    geceSayisi: number;
    kisiSayisi: number;

    rezervasyonDurumu: string;
    rezervasyonDurumuLabel: string;

    odaNolari: string[];
    odaTipleri: string[];
}

export interface RezervasyonDurumDagilimiRaporDto {
    tesisId: number;
    tesisAdi: string | null;
    baslangic: string;
    bitis: string;

    odaTipiId: number | null;
    odaTipiAdi: string | null;
    durum: string | null;
    durumLabel: string | null;

    baslik: string;

    ozet: RezervasyonDurumDagilimiOzetDto;
    durumlar: RezervasyonDurumDagilimiDurumSatiriDto[];
    odaTipleri: RezervasyonDurumDagilimiOdaTipiSatiriDto[];
    rezervasyonlar: RezervasyonDurumDagilimiRezervasyonDto[];
}

File: backend/Raporlar/RezervasyonDurumDagilimi/Dto/RezervasyonDurumDagilimiRaporDto.cs
namespace STYS.Raporlar.RezervasyonDurumDagilimi.Dto;

public class RezervasyonDurumDagilimiRaporDto
{
    public int TesisId { get; set; }
    public string? TesisAdi { get; set; }
    public DateTime Baslangic { get; set; }
    public DateTime Bitis { get; set; }

    public int? OdaTipiId { get; set; }
    public string? OdaTipiAdi { get; set; }
    public string? Durum { get; set; }
    public string? DurumLabel { get; set; }

    public string Baslik { get; set; } = "";

    public RezervasyonDurumDagilimiOzetDto Ozet { get; set; } = new();
    public List<RezervasyonDurumDagilimiDurumSatiriDto> Durumlar { get; set; } = [];
    public List<RezervasyonDurumDagilimiOdaTipiSatiriDto> OdaTipleri { get; set; } = [];
    public List<RezervasyonDurumDagilimiRezervasyonDto> Rezervasyonlar { get; set; } = [];
}

public class RezervasyonDurumDagilimiOzetDto
{
    public int ToplamRezervasyonSayisi { get; set; }

    public int TaslakSayisi { get; set; }
    public int OnayliSayisi { get; set; }
    public int CheckInTamamlandiSayisi { get; set; }
    public int CheckOutTamamlandiSayisi { get; set; }
    public int IptalSayisi { get; set; }

    public int GerceklesenRezervasyonSayisi { get; set; }
    public int DevamEdenRezervasyonSayisi { get; set; }

    public decimal IptalOrani { get; set; }
    public decimal GerceklesmeOrani { get; set; }
    public decimal CheckInOrani { get; set; }
    public decimal CheckOutOrani { get; set; }

    public int ToplamKisiSayisi { get; set; }
    public int ToplamGeceSayisi { get; set; }
}

public class RezervasyonDurumDagilimiDurumSatiriDto
{
    public string Durum { get; set; } = "";
    public string DurumLabel { get; set; } = "";
    public int RezervasyonSayisi { get; set; }
    public int KisiSayisi { get; set; }
    public int GeceSayisi { get; set; }
    public decimal Oran { get; set; }
}

public class RezervasyonDurumDagilimiOdaTipiSatiriDto
{
    public int OdaTipiId { get; set; }
    public string OdaTipiAdi { get; set; } = "";
    public int RezervasyonSayisi { get; set; }
    public int IptalSayisi { get; set; }
    public int GerceklesenSayisi { get; set; }
    public int KisiSayisi { get; set; }
    public int GeceSayisi { get; set; }
    public decimal IptalOrani { get; set; }
    public decimal GerceklesmeOrani { get; set; }
}

public class RezervasyonDurumDagilimiRezervasyonDto
{
    public int RezervasyonId { get; set; }
    public string? ReferansNo { get; set; }
    public string? MisafirAdiSoyadi { get; set; }

    public DateTime GirisTarihi { get; set; }
    public DateTime CikisTarihi { get; set; }

    public int GeceSayisi { get; set; }
    public int KisiSayisi { get; set; }

    public string RezervasyonDurumu { get; set; } = "";
    public string RezervasyonDurumuLabel { get; set; } = "";

    public List<string> OdaNolari { get; set; } = [];
    public List<string> OdaTipleri { get; set; } = [];
}
